### DISCUSSION TITLE
Covariance Matrix for pestpp-ies

### DISCUSSION THREAD

**Discussion by d-strom** [pypest/pyemu#494] [Q&A] [answered]
Hi,

I have been trying to make use of a geospatial covariance matrix for prior monte carlo following some examples in the gmdsi tutorial notebooks.

I notice however that in the examples there is one cov-file supplied whereas the plproc workflow or if I try to generate a cov matrix form a .tpl-file generate one cov Matrix per parameter group.

Is there an example workflow on how to create a single covariance Matrix for multiple parameter groups or how to do a pest-ies run with noptmax set to -1 while supplying multiple covariance matrices for each involved parameter group?

also, having pilot point supplied with a dense pattern and manual points in various locations gives me covariance matrices that look rather strange when plotted. Nothing like the very symetrical and smooth examples in the tutorials.

Is the connection between the pilot points from my pst file neccesary? What purpose does it have? Or can you simply generate a gemeric covariance matrix specifying the distance and relationship parameters should have in space?

sincerely
David

**Comment by jtwhite79** [comment, marked as answer]
you can pass a pest-style .unc (uncertainty) file to the parcov argument if you want to have a cov matrix for each par group.   the pattern in the cov matrix may not look like the ones in the tutorials bc its a function of the pilot point spacing and the geostruct.  However unless you are using geostatistical hyperparameters, I think the cov matrix should still symmetric about the diagonal...

**Reply by d-strom** [reply]
Thanks Jeremy,

Sounds like that should solve my problem.
Perhaps my description was a bit off target.. my matrices are indeed symetrical around the diagonal. Its just that they have a more, in lack off better words, heterogenious look on either side of the diagonal than in the examples. Might still be in order i suppose given my pilot point pattern.

sincerely
David

**Comment by d-strom** [comment]
@jtwhite79 if I may add to this question something I have been wondering regarding the covariance matrices.
When i pass an .unc-file containing all the covariance matrices (one per par group), pestpp informs me that I'm missing the names of the parameters in the file. I am thus forced to add them like in the picture below.

Could you elaborate on the role of the matrices vs the parameters mentioned in the standard_deviation section? does this mean that pestpp is not able to pair the matrices to the parameters and thus is asking me to express it with a std instead?
If so, do the cov-matrices fill a function at all? I was under the impression that they will guide my model with the geostatistical patterns that I have built in order to make the calibrated parameter field vary in a somewhat resonable way. I created them from model muse with plproc, mkppstat and ppcov_sva from the pest gw-utils. 

![image](https://github.com/user-attachments/assets/712313d3-88c3-4ee0-a7b9-f0dc97c8fba0)


//David

**Reply by jtwhite79** [reply]
The unc must name all adjustable parameters - i think pest_hp also requires this...the implication is that those stdev pars are statistically independent of all the other parameters.  But the covariance blocks still describe correlation between parameters in each respective block.  pestpp-ies uses that unc file to build a block-diagonal cov matrix and then draws from it to form the prior par ensemble so that the prior ensemble reflects the variance and covariances in the unc file

**Reply by d-strom** [reply]
Right! So that is just the requirement of the file format? it really doesnt influence the block diagonal cov-matrices and how they behave in the inversion process?

thanks

**Reply by jtwhite79** [reply]
well it influences the block-diagonal matrix in as much as those stdev parameters only have diagonal entries and therefore uncorrelated with any other parameter...

**Reply by d-strom** [reply]
got it! thanks
